src/Unused/Cache/Fingerprint.hs

```diff
@@ -0,0 +1,25 @@
+module Unused.Cache.Fingerprint
+    ( sha
+    ) where
+
+import System.Process
+
+sha :: IO String
+sha =
+    getSha <$> (fileList >>= md5Input >>= sortInput >>= md5Result)
+  where
+    getSha = head' . lines
+    head' (x:_) = x
+    head' _ = ""
+
+fileList :: IO String
+fileList = readProcess "find" [".", "-type", "f", "-not", "-path", "*/tmp/unused/*", "-print0"] ""
+
+md5Input :: String -> IO String
+md5Input = readProcess "xargs" ["-0", "-P", "1", "md5"]
```

Comment:
I means to pass the files as multiple arguments to one invocation of the
command (like when piping to xargs). This is different than the semicolon
invocation which calls the command once for each found file.

On Sun, May 15, 2016, 14:02 Gabe Berke-Williams <notifications@github.com>
wrote:

> In src/Unused/Cache/Fingerprint.hs
> <https://github.com/joshuaclayton/unused/pull/29#discussion_r63295603>:
>
> > +
> > +import System.Process
> > +
> > +sha :: IO String
> > +sha =
> > +    getSha <$> (fileList >>= md5Input >>= sortInput >>= md5Result)
> > +  where
> > +    getSha = head' . lines
> > +    head' (x:_) = x
> > +    head' _ = ""
> > +
> > +fileList :: IO String
> > +fileList = readProcess "find" [".", "-type", "f", "-not", "-path", "*/tmp/unused/*", "-print0"] ""
> > +
> > +md5Input :: String -> IO String
> > +md5Input = readProcess "xargs" ["-0", "-P", "1", "md5"]
>
> Random shell question -- what does the + in find -exec {} + mean?
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/joshuaclayton/unused/pull/29/files/b589d0b0a1f9c6948e2878750e24fe075be39bfe#r63295603>
>